tmc-langs-ant/src/main/java/fi/helsinki/cs/tmc/langs/ant/AntPlugin.java

```diff
@@ -21,23 +32,164 @@ public String getLanguageName() {
     }
 
     @Override
-    public ImmutableList<Path> findExercises(Path basePath) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    public ExerciseDesc scanExercise(Path path, String exerciseName) {
+        if (!isExerciseTypeCorrect(path)) {
+            return null;
+        }
+
+        List<String> output = startProcess(buildTestScannerArgs(path, null));
+        String outputString = "";
+        for (String line : output) {
+            outputString += line;
+        }
+        return resultParser.parseScannerOutput(outputString, exerciseName);
     }
 
     @Override
-    public ExerciseDesc scanExercise(Path path, String exerciseName) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    protected boolean isExerciseTypeCorrect(Path path) {
+        return new File(path.toString() + File.separatorChar + "build.xml").exists();
     }
 
     @Override
     public RunResult runTests(Path path) {
-        ArrayList<String> args = new ArrayList<>();
-        throw new UnsupportedOperationException("Not supported yet.");
+        if (!buildAntProject(path)) {
+            return buildRunResult;
+        }
+
+        List<String> runnerArgs = buildTestRunnerArgs(path);
+        startProcess(runnerArgs);
+
+        File resultFile = new File(path.toString() + resultsFile);
+        RunResult result = resultParser.parseTestResult(resultFile);
+        resultFile.delete();
+
+        return result;
     }
 
-    @Override
-    public boolean isExerciseTypeCorrect(Path path) {
-        return new File(path.toString() + File.separatorChar + "build.xml").exists();
+    /**
+     * Runs the build.xml file for the the given exercise.
+     *
+     * @param path The file path of the exercise directory.
+     * @return true if build success, else return false.
+     */
+    @VisibleForTesting
+    protected boolean buildAntProject(Path path) {
+        File buildFile = new File(path.toString() + File.separatorChar + "build.xml");
+        Project buildProject = new Project();
+        buildProject.setUserProperty("ant.file", buildFile.getAbsolutePath());
+        buildProject.setProperty("javac.fork", "true");
+        buildProject.init();
+        buildProject.setBaseDir(path.toAbsolutePath().toFile());
+        File buildLog;
+        
+        try {
+            
+            DefaultLogger logger = new DefaultLogger();
+            buildLog = new File(path.toString(), "build_log.txt");
```

Comment:
Maybe the build_log.txt and build.xml should be in private static final variables?